compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -2431,7 +2431,31 @@ TR::Register *OMR::X86::TreeEvaluator::andORStringEvaluator(TR::Node *node, TR::
    return resultReg;
    }
 
-TR::Register *OMR::X86::TreeEvaluator::OverflowCHKEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+TR::Block *OMR::X86::TreeEvaluator::GetOverflowCatchBlock(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   //make sure the overflowCHK has a catch block first
+   TR::Block *overflowCatchBlock = NULL;
+   TR::list<TR::CFGEdge*> excepSucc =cg->getCurrentEvaluationTreeTop()->getEnclosingBlock()->getExceptionSuccessors();
+   for (auto e = excepSucc.begin(); e != excepSucc.end(); ++e) 
+      {    
+      TR::Block *dest = toBlock((*e)->getTo());
+      if (dest->getCatchBlockExtension()->_catchType == TR::Block::CanCatchOverflowCheck)
+            overflowCatchBlock = dest;
+      }    
+   TR_ASSERT(overflowCatchBlock != NULL, "OverflowChk node %p doesn't have overflow catch block\n", node);
+
+   //the BBStartEvaluator will generate the label but in this case the catch block might not been evaluated yet
+   TR::Node * bbstartNode = overflowCatchBlock->getEntry()->getNode();
+   TR_ASSERT((bbstartNode->getOpCodeValue() == TR::BBStart), "catch block entry %p must be TR::BBStart\n", bbstartNode);
+   if (!bbstartNode->getLabel())
+      {
+      TR::LabelSymbol *label = generateLabelSymbol(cg);
+      bbstartNode->setLabel(label);
+      }
+   return overflowCatchBlock;
+   }
+
+void OMR::X86::TreeEvaluator::GenArithmeticInstructions(TR::Node *node, TR::CodeGenerator *cg)
```

Comment:
The fact that this takes an OverflowCHK / UnsignedOverflowCHK node is surprising given the node - I think this should have a better name and assertion to check the opcode of node is what is expected.